aggregator/src/main/java/com/redhat/cloud/notifications/db/EmailAggregationRepository.java

```diff
@@ -23,13 +24,15 @@ public class EmailAggregationRepository {
     EntityManager entityManager;
 
     public List<AggregationCommand> getApplicationsWithPendingAggregationAccordinfOrgPref(LocalDateTime end) {
+        // Must takes every EmailAggregation supposed to be processed on last 15 min
+        // it covers cases when aggregation job may be run with few minutes late (ie: 05:01, 07,32)
         String query = "SELECT DISTINCT ea.orgId, ea.bundleName, ea.applicationName, acp.lastRun FROM EmailAggregation ea, AggregationOrgConfig acp WHERE " +
             "ea.orgId = acp.orgId AND ea.created > acp.lastRun AND ea.created <= :end " +
-            "AND hour(acp.scheduledExecutionTime) = :runningHour";
-
+            "AND :endTime >= acp.scheduledExecutionTime AND (:endTime - acp.scheduledExecutionTime) < CAST(:cutoff as LocalTime)";
         Query hqlQuery = entityManager.createQuery(query)
-                .setParameter("runningHour", end.getHour())
-                .setParameter("end", end);
+                .setParameter("endTime", end.toLocalTime())
```

Comment:
Since `end` is actually `now`, we could name everything like this:
```suggestion
                .setParameter("now", now.toLocalTime())
```